### PR TITLE
Handle more cases in F1 help service

### LINKED ISSUE #51878: F1 service redirects to latest visited F1 keyword when none is available
- Open a source code and press F1 in any well-known keyword (e.g public)
- It redirects to the correct page
- Now, press F1 on `:` in something like `MyType : IMyType`
- It redirects to `public` docs page

Doing "nothing" appears to be less confusing.

cc @davidwengier

### LINKED ISSUE #60154: F1 service redirects to binary operator for `>` in type argument list (generic method invocation)
**Version Used**: 

main

**Steps to Reproduce**:

```csharp
        [Fact, Trait(Traits.Feature, Traits.Features.F1Help)]
        public async Task TestGenericAngle2()
        {
            await TestAsync(
@"class Program
{
    static void generic<T>(T t)
    {
        generic<int[|>|](0);
    }
}", "Program.generic``1");
        }
```

Note: The above test will fail because actual is `>_CSharpKeyword`. This doesn't align with `TestGenericAngle` which expects:

> "Program.generic``1"

IMO both results are not good. We should redirect to https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/types/generics. The page doesn't yet have an F1 keyword.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #60154
Fixes #51878

- Handles different meanings of EqualsToken (part of https://github.com/dotnet/roslyn/issues/48392).

- Handles `yield return` and `yield break` (https://github.com/dotnet/roslyn/issues/23881#issuecomment-353659895)

docs PR: https://github.com/dotnet/docs/pull/28653 https://github.com/dotnet/docs/pull/28660

@davidwengier @CyrusNajmabadi Can you take a look? Thanks!

### CODE REVIEW

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 1]
```diff
@@ -239,6 +232,12 @@ private static bool TryGetTextForOperator(SyntaxToken token, Document document,
                 return true;
             }
 
+            if (token.IsKind(SyntaxKind.LessThanToken, SyntaxKind.GreaterThanToken) && token.Parent.IsKind(SyntaxKind.TypeParameterList, SyntaxKind.TypeArgumentList))
+            {
+                text = Keyword("generics");
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:237]: I'm using the `Keyword` call for consistency, but let me know if I shouldn't use it in this case.

**Review by Youssef1313 — COMMENTED** [review]
@RikkiGibson FYI we will need to handle `!!`. But it's waiting on https://github.com/dotnet/docs/issues/28443. Do you want a tracking issue for this?

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 2]
```diff
@@ -50,14 +51,15 @@ public override async Task<string> GetHelpTermAsync(Document document, TextSpan
                 var semanticModel = await document.ReuseExistingSpeculativeModelAsync(span, cancellationToken).ConfigureAwait(false);
 
                 var result = TryGetText(token, semanticModel, document, cancellationToken);
-                if (string.IsNullOrEmpty(result))
+                if (result is null)
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:54]: why this cahnge?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 3]
```diff
@@ -227,6 +224,79 @@ private static bool TryGetTextForOperator(SyntaxToken token, Document document,
...
+                text = Keyword("=");
+                return true;
+            }
+
+            if (token.IsKind(SyntaxKind.LessThanToken, SyntaxKind.GreaterThanToken))
+            {
+                if (token.Parent.IsKind(SyntaxKind.FunctionPointerParameterList))
+                {
+                    // PROTOTYPE: Figure out where to redirect.
+
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:296]: @BillWagner @333fred Is there a documentation page we can redirect to here?

**Comment by RikkiGibson** [comment]
> @RikkiGibson FYI we will need to handle `!!`. But it's waiting on [dotnet/docs#28443](https://github.com/dotnet/docs/issues/28443). Do you want a tracking issue for this?

No thanks, I'll just subscribe to the docs issue.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs (hunk 3 above)
Comment (BillWagner) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:296]: Not yet. I'll ping you and @RikkiGibson on the docs issue referenced below when I open that PR.

**Review by davidwengier — APPROVED** [review]

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs (hunk 3 above)
Comment (Youssef1313) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:296]: @BillWagner Sorry for not clarifying. This was about function pointers

https://github.com/dotnet/roslyn/blob/64c5fd150f87ddea7a5fa673555ea2ad74a75e58/src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs#L291-L295

I found now that https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/unsafe-code is the most suitable page.

Opened https://github.com/dotnet/docs/pull/28669.

**Pull request merged**